pipeline-scripts/baseline.groovy

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env groovy
+
+def pipeline_id = env.BUILD_ID
+def node_label = NODE_LABEL.toString()
+def baseline = BASELINE_SCALE_TEST.toString().toUpperCase()
+def property_file_name = "baseline.properties"
+
+println "Current pipeline job build id is '${pipeline_id}'"
+
+// run baseline scale test
+stage ('baseline_scale_test') {
+	if (baseline == "TRUE") {
+		currentBuild.result = "SUCCESS"
+		node(node_label) {
+			// get properties file
+			if (fileExists(property_file_name)) {
+				println "Looks like the property file already exists, erasing it"
+				sh "rm ${property_file_name}"
+			}
+			// get properties file
+			sh "wget ${BASELINE_PROPERTY_FILE} -O ${property_file_name}"
+			sh "cat ${property_file_name}"
+			def baseline_properties = readProperties file: property_file_name
+			def jump_host = baseline_properties['JUMP_HOST']
+			def user = baseline_properties['USER']
+			def use_proxy = baseline_properties['USE_PROXY']
+			def proxy_user = baseline_properties['PROXY_USER']
+			def proxy_host = baseline_properties['PROXY_HOST']
+			def token = baseline_properties['GITHUB_TOKEN']
+			def repo = baseline_properties['PERF_REPO']
+			def server = baseline_properties['PBENCH_SERVER']
+	
+			// debug info
+			println "----------USER DEFINED OPTIONS-------------------"
+			println "-------------------------------------------------"
+			println "-------------------------------------------------"
+			println "JUMP_HOST: '${jump_host}'"
+			println "USER: '${user}'"
+			println "USE_PROXY: '${use_proxy}'"
+			println "PROXY_USER: '${proxy_user}'"
+			println "PROXY_HOST: '${proxy_host}'"
+			println "TOKEN: '${token}'"
+			println "-------------------------------------------------"
+			println "-------------------------------------------------"
+
+			// copy the parameters file to jump host
+			sh "git clone https://${token}@${repo} ${WORKSPACE}/perf-dept && chmod 600 ${WORKSPACE}/perf-dept/ssh_keys/id_rsa_perf"
+			sh "scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i ${WORKSPACE}/perf-dept/ssh_keys/id_rsa_perf ${property_file_name} root@${jump_host}:/root/properties"
```

Comment:
I don't think it is a good idea to have the groovy script executing cli commands as we have a bash build section already.